### PR TITLE
feat(singly): UnsafeRef impl

### LINKED ISSUE #37: singly-linked list impls for `UnsafeRef`
follow-on from #8.

this is going to look pretty similar to the code I added in a15318754af30f1ffc64c1f40a8f77d217ad324d.

### PR DESCRIPTION
**Pull request by amanjeev**
resolves https://github.com/sos-os/alarm/issues/37

### CODE REVIEW

**Comment by amanjeev** [comment]
I actually think some naming is bad in this one, esp. `extend`.

**Comment by amanjeev** [comment]
Why isn't travis build running for this PR. 🤔

**Comment by hawkw** [comment]
@amanjeev I suspect the GitHub/Travis webhooks integration got messed up somehow when I moved the repo to the `sos-os` organization; I'll look at the settings and see if I can figure that out.

**Comment by hawkw** [comment]
Okay, I went into the repo settings and sent Travis a test webhook; that seems to have gotten it unstuck and it looks like [your build](https://www.travis-ci.org/sos-os/alarm/builds/337310115) has passed! ✅

**Review by hawkw — CHANGES_REQUESTED** [review]
The code looks good to me! Before merging, can I get you to make the documentation change I mentioned in https://github.com/sos-os/alarm/issues/37#issuecomment-362938782?

**Comment by hawkw** [comment]
@amanjeev 
> I actually think some naming is bad in this one, esp. `extend`.

I'm not sure which naming you mean here --- if you're saying that the name of the `extend` method isn't ideal for a stack, that's a fair point, but since `Extend` is a standard-library trait that we're implementing here, we can't change the name of the method.

File: intruder_alarm/src/singly/mod.rs [hunk 1]
```diff
@@ -20,11 +20,14 @@ mod tests;
 // Public API types
 //-----------------------------------------------------------------------------
 //  List
-/// An intrusive singly-linked list.
+/// An intrusive singly-linked list mainly for stack operations.
```
Comment (hawkw) [inline intruder_alarm/src/singly/mod.rs]: Honestly, we could probably change this line to just
```rust
/// An intrusive singly-linked stack.
```

If you think it makes sense, I'd be open to even just renaming this type to `Stack` rather than `List`, too; might express the intended use more clearly. I'd make that change in a separate PR though.

**Review by hawkw — DISMISSED** [review]
LGTM, thanks for the docs change! 👍

File: intruder_alarm/src/singly/mod.rs (hunk 1 above)
Comment (amanjeev) [reply intruder_alarm/src/singly/mod.rs]: OK. That makes much more sense (both, the line here and renaming to Stack).

**Comment by amanjeev** [comment]
> if you're saying that the name of the extend method isn't ideal for a stack

That's what I meant but thanks for correcting me.

**Review by hawkw — APPROVED** [review]
looks good to me! :tada:
